### PR TITLE
ORB-541 - M.9 - Empty data doesn’t always revert

### PR DESCRIPTION
**Pull request by AlbyGiaco**

### CODE REVIEW

File: contracts/modules/AutoCompoundModule.sol [hunk 1]
```diff
@@ -58,6 +60,7 @@ contract AutoCompoundModule is BaseModule {
         );
 
         (, bytes32 data) = IPositionManager(positionManagerAddress).getModuleInfo(tokenId, address(this));
+        require(data != bytes32(0), 'AutoCompoundModule::_checkIfCompoundIsNeeded: module data cannot be empty');
```
Comment (RiccardoGalbusera) [inline contracts/modules/AutoCompoundModule.sol:63]: Is this require really necessary? Didn't we already make sure that module data are not empty?

File: contracts/modules/IdleLiquidityModule.sol [hunk 2]
```diff
@@ -35,6 +41,7 @@ contract IdleLiquidityModule is BaseModule {
     {
         uint24 tickDistance = _checkDistanceFromRange(tokenId);
         (, bytes32 rebalanceDistance) = positionManager.getModuleInfo(tokenId, address(this));
+        require(rebalanceDistance != bytes32(0), 'IdleLiquidityModule:: rebalance: Rebalance distance is 0');
```
Comment (RiccardoGalbusera) [inline contracts/modules/IdleLiquidityModule.sol:44]: same for this require

File: contracts/modules/AutoCompoundModule.sol (hunk 1 above)
Comment (luduvigo) [reply contracts/modules/AutoCompoundModule.sol:63]: where exactly @RiccardoGalbusera ?

Comment (AlbyGiaco) [reply contracts/modules/AutoCompoundModule.sol:63]: DoubleCheck, cause the data is passing in storage and we are not sure what we get

**Review by luduvigo — APPROVED** [review]

**Pull request merged**